easybuild/easyblocks/a/anaconda.py

```diff
@@ -67,83 +68,60 @@ def post_install_step(log, installdir, post_install_cmd):
         run_cmd(post_install_cmd, log_all=True, simple=True)
         log.info('Post command run {}'.format(post_install_cmd))
 
+
 def initialize_conda_env(installdir):
     """ Initialize the conda env """
-
     rmtree2(installdir)
     cmd = "conda config --add create_default_packages setuptools"
     run_cmd(cmd, log_all=True, simple=True)
 
-class EB_anaconda(EasyBlock):
-    """Support for building/installing anaconda."""
+
+class EB_Anaconda(Binary):
+    """Support for building/installing Anaconda."""
 
     @staticmethod
     def extra_options(extra_vars=None):
-        """Extra easyconfig parameters specific to EB_anaconda easyblock."""
-        extra_vars = EasyBlock.extra_options(extra_vars)
+        """Extra easyconfig parameters specific to Anaconda."""
+        extra_vars = Binary.extra_options(extra_vars)
         extra_vars.update({
             'pre_install_cmd': [None, "Commands before install: setting custom environmental variables, etc", CUSTOM],
             'post_install_cmd': [None, "Commands after install: pip install, cpanm install, etc", CUSTOM],
         })
         return extra_vars
 
-    def __init__(self, *args, **kwargs):
-        """Initialize EB_anaconda-specific variables."""
-
-        super(EB_anaconda, self).__init__(*args, **kwargs)
-
-    def extract_step(self):
-        """Move all source files to the build directory"""
-
-        self.src[0]['finalpath'] = self.builddir
-
-        # copy source to build dir.
-        for source in self.src:
```

Comment:
Will this remove the extract_step for all the blocks? In CondaEnv I am sometimes pulling from a remote environment, which means there is no file source. Will this have an effect if there is no source?